### PR TITLE
add iter_content retry

### PR DESCRIPTION
**Pull request by obi-t4**
# About this PR.
This PR fixes issue when `response.iter_content` raises timeout error, Retry function doesn't retry.
In the previous implementation, retry is done manually so iter_content timeout was handled, but the current implementation is not.

`https://github.com/tier4/ota-client/pull/115?w=1` is easer to see diff.

# issue
https://tier4.atlassian.net/browse/T4PB-15676

### CODE REVIEW

File: app/ota_client.py [hunk 1]
```diff
@@ -113,36 +114,50 @@ def __call__(
...
+                        for data in response.iter_content(chunk_size=self.CHUNK_SIZE):
+                            hash_f.update(data)
+                            f.write(data)
+                        break
+                    except Exception:  # catch all exceptions
+                        error_count += 1
+                        if error_count > self.RETRY_COUNT:
+                            logger.exception("error: iter_content")
+                            raise
+                        time.sleep(self.BACKOFF_MAX)
```
Comment (Bodong-Yang) [inline app/ota_client.py:142]: If I understand correctly, here we sleep `self.BACKOFF_MAX` for every retry, is that right?

Comment (obi-t4) [reply app/ota_client.py:142]: Yes, when iter_content raises, sleep self.BACKOFF_MAX(=10s) every time.

Comment (Bodong-Yang) [reply app/ota_client.py:142]: Is it too long for the first retry on `iter_content`? Or should we use a backoff function to generate the for every retry delay here?

Comment (obi-t4) [reply app/ota_client.py]: get_backoff_time has been implemented.

File: app/ota_client.py [hunk 2]
```diff
@@ -107,42 +109,61 @@ def _path_to_url(base: str, p: Union[Path, str]) -> str:
...
-            raise OtaErrorRecoverable(msg)
-
-        # prepare hash
-        hash_f = sha256()
-        with open(dst, "wb") as f:
-            for data in response.iter_content(chunk_size=self.CHUNK_SIZE):
-                hash_f.update(data)
-                f.write(data)
+
+        while True:
```
Comment (Youneko) [inline app/ota_client.py:123]: [IMO] Retry()によるリトライでリトライされていないパターンを補っているように見えました。[retrying module](https://github.com/rholder/retrying)を使ってリトライ処理を1本にした方が良いように思いましたがどうでしょうか？

**Review by pga2rn — APPROVED** [review]
I think it should work as expected!

**Review by pga2rn — APPROVED** [review]

File: app/ota_client.py (hunk 2 above)
Comment (pga2rn) [reply app/ota_client.py:123]: The module you referenced is no-longer maintained.😥
And here we already use the urllib3.retry to handle connection problem. But it seems sometime there are timeout problem in iter_content, so another simple retry implemented in this PR to re-do connection with urllib3.retry I think.

**Comment by obi-t4** [comment]
@Bodong-Yang @Youneko 
We have plans to refactor this function, so I'll close this PR.

**Pull request closed without merging**